orthoinference/src/main/java/org/reactome/orthoinference/InferEvents.java

```diff
@@ -0,0 +1,307 @@
+				}
+			} else {
+				InferEWAS.updateRefDb();
+			}
+			InferEvents.createSpeciesInst(speciesName);
+			OrthologousEntity.setSpeciesInst(speciesInst);
+			InferEWAS.setSpeciesInst(speciesInst);
+			GenerateInstance.setSpeciesInst(speciesInst);
+			InferEvents.setSummationInst();
+			InferEvents.setEvidenceTypeInst();
+			OrthologousEntity.setComplexSummationInst();
+
+/**
+ *  Start of ReactionlikeEvent inference. Retrieves all human ReactionlikeEvents, and attempts to infer each for the species.
+ */
+		// Gets DB instance of source species (human)
+		Collection<GKInstance> sourceSpeciesInst = (Collection<GKInstance>) dbAdaptor.fetchInstanceByAttribute("Species", "name", "=", speciesToInferFromLong);
+		if (!sourceSpeciesInst.isEmpty())
+		{
+			String dbId = sourceSpeciesInst.iterator().next().getDBID().toString();
+			// Gets Reaction instances of source species (human)
+			Collection<GKInstance> reactionInstances = (Collection<GKInstance>) dbAdaptor.fetchInstanceByAttribute("ReactionlikeEvent", "species", "=", dbId);
+
+			ArrayList<Long> dbids = new ArrayList<Long>();
+			HashMap<Long, GKInstance> reactionMap = new HashMap<Long, GKInstance>();
+			for (GKInstance reactionInst : reactionInstances) {
+				dbids.add(reactionInst.getDBID());
+				reactionMap.put(reactionInst.getDBID(), reactionInst);
+			}
+			// For now sort the instances by DB ID so that it matches the Perl sequence
+			Collections.sort(dbids);
+			
+			if (!reactionInstances.isEmpty())
+			{
+				for (Long dbid : dbids)
+				{
+					GKInstance reactionInst = reactionMap.get(dbid);
+					// Check if the current Reaction already exists for this species, that it is a valid instance (passes some filters), and that it doesnt have a Disease attribute.
+					// Adds to manualHumanEvents array if it passes conditions. This code block allows you to re-run the code without re-inferring instances.
+					ArrayList<GKInstance> previouslyInferredInstances = new ArrayList<GKInstance>();
+					for (GKInstance orthoEventInst : (Collection<GKInstance>) reactionInst.getAttributeValuesList(ReactomeJavaConstants.orthologousEvent))
+					{
+						GKInstance reactionSpeciesInst = (GKInstance) orthoEventInst.getAttributeValue(ReactomeJavaConstants.species);
+						if (reactionSpeciesInst.getDBID() == speciesInst.getDBID() && orthoEventInst.getAttributeValue(ReactomeJavaConstants.isChimeric) == null)
+						{
+							previouslyInferredInstances.add(orthoEventInst);
+							
+						}
+					}
+					for (GKInstance inferredFromInst : (Collection<GKInstance>) reactionInst.getAttributeValuesList(ReactomeJavaConstants.inferredFrom))
+					{
+						GKInstance reactionSpeciesInst = (GKInstance) inferredFromInst.getAttributeValue(ReactomeJavaConstants.species);
+						if (reactionSpeciesInst.getDBID() == speciesInst.getDBID() && inferredFromInst.getAttributeValue(ReactomeJavaConstants.isChimeric) == null)
+						{
+							previouslyInferredInstances.add(inferredFromInst);
+						}
+					}
```

Comment:
This loop and the previous loop can be extracted into a common method passing in either `ReactomeJavaConstants.orthologousEvent` or `ReactomeJavaConstants.inferredFrom`.